### DISCUSSION TITLE
About GiveWP Stubs

### DISCUSSION THREAD

**Discussion by FikretCin** [impress-org/givewp#7154] [Developer Q&A]
Hello im WordPress payment gateway plugin developer. I use tools such as phpstan and phpcs to maintain code quality while coding my plugin. phpstan have undefined function and class errors because i haven't givewp stubs package like them.

[WordPress Stubs](https://github.com/php-stubs/wordpress-stubs)
[WooCommerce Stubs](https://github.com/php-stubs/woocommerce-stubs)

If you have such a package, can you send it to me?

Thanks for answer, regards.

**Comment by jonwaldstein** [comment]
Hi @FikretCin 

I don't believe we have any stubs.

If you are planning on building a gateway, I encourage you to use the [GiveWP payment gateway api docs](https://givewp.com/documentation/developers/how-to-build-a-gateway-add-on-for-givewp/) which should be more phpstan forgiving as it uses OOP and mostly follows psr-12
